### ISSUE TITLE
Slash in category name

### ISSUE THREAD

**Issue by pointia** [techdivision/import#178]
Hi Tim,

i tried to import with this awesome framework and figured out that importing categories with a forward slash is not working.

The category processor always tries to create additional categories.

I'v written 2 patches that solve the problem. Could you please test them and integrate in your framework.

[techdivision.zip](https://github.com/techdivision/import/files/4589351/techdivision.zip)

The third patch is for msi if the msi tables do not exist.

**Comment by wagnert** [comment]
Hi @pointia Thanks for using M2IF and reporting :-) Would it be possible for you to create a PR, that would make things much easier?

**Comment by pointia** [comment]
I didn't create a PR because it looks like that several modules are involved at least techdivision/import , techdivision/import-converter-product-category and possibly techdivision/import-product-url-rewrite (at least I'm getting an error when importing such categories and I did not fix that yet)

I thought, in that case, it would be better to provide a file with patch files instead

**Comment by wagnert** [comment]
@pointia All right, I'll check it as soon as possible and keep you updated :-)

**Comment by wagnert** [comment]
Hi @pointia I've two more questions:
# I don't understand the relation of the `try {} catch () {}` block in the [ProductSourceItemObserver](https://github.com/techdivision/import-product-msi/blob/15.x/src/Observers/ProductSourceItemObserver.php#L85) with this issue?
# Is it possible, that you send/attach a example file that we can use to reproduce the issue?

Thanks again, for supporting!

**Comment by pointia** [comment]
Hi @wagnert,

thanks for keeping track of this issue.

1) This is not related to the issue. But this is needed when MSI tables are not present. So actually this should be in another issue.
2) Please find attached a sample

[product.zip](https://github.com/techdivision/import/files/4595234/product.zip)

**Comment by wagnert** [comment]
@pointia Thanks for the CSV file, I'll check this as soon as possible. Regarding the MSI tables, I don't understand this, because by default the MSI functionality should not be executed in versions that doesn't provide this functionality. So this should never happen? Or did I understand this wrong? :-)

**Comment by pointia** [comment]
@wagnert 
Thanks a lot.
Regardin MSI: I also thought so but it looks like as if the observer is called from the base import module even though MSI is disabled.

**Comment by wagnert** [comment]
@pointia This should not happen, if the correct Magento version has been specified, but I'll check this again to make sure, that this will not happen. Thanks again for reporting!! :-)

**Comment by wagnert** [comment]
Hi @pointia, we've fixed the issue, but to be compliant with the quoting/escaping in the category import we've decided to go another way. We've created an [example file](https://github.com/techdivision/import-sample-data/blob/4.x/generic/data/products/category-create/product-import_20200602-121056_01.csv) that show's you the expected quoting/escaping for categories within a CSV file containing product data. Feel free to ask, if you've any further questions. And thanks again for reporting 👍
